### PR TITLE
[MM-57716] Convert ./components/setting_item.tsx from Class Component to Function Component 

### PR DESCRIPTION
**Pull request by Camillarhi**
#### Summary

Convert `./components/setting_item.tsx` from class to function component

#### Ticket Link

Fixes https://github.com/mattermost/mattermost/issues/26684
Jira https://mattermost.atlassian.net/browse/MM-57716

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: webapp/channels/src/components/setting_item.tsx [hunk 1]
```diff
@@ -41,37 +41,42 @@ type Props = {
...
 
-export default class SettingItem extends React.PureComponent<Props> {
-    minRef: RefObject<SettingItemMinComponent>;
-
-    constructor(props: Props) {
-        super(props);
-
-        this.minRef = React.createRef();
-    }
+const SettingItem: FC<Props> = ({
```
Comment (larkox) [inline webapp/channels/src/components/setting_item.tsx:44]: Any strong reason to be explicit about the type? In general is not needed and we don't add it, so curious if you have any strong opinion about it.

**Review by larkox — APPROVED** [review]
Great work! Just one question

File: webapp/channels/src/components/setting_item.tsx (hunk 1 above)
Comment (Camillarhi) [reply webapp/channels/src/components/setting_item.tsx:44]: No reason. I can change the component to be this 
```
{
    active,
    areAllSectionsInactive,
    section,
    max,
    updateSection,
    title,
    isDisabled,
    describe = '',
    collapsedEditButtonWhenDisabled = false,
}: Props
```

**Review by devinbinnie — APPROVED** [review]

**Comment by devinbinnie** [comment]
@Camillarhi Can you have a look at the failing tests?

File: webapp/channels/src/components/setting_item.tsx (hunk 1 above)
Comment (larkox) [reply webapp/channels/src/components/setting_item.tsx:44]: Yeah, just to be uniform with other components, let's remove the typing.

Comment (Camillarhi) [reply webapp/channels/src/components/setting_item.tsx]: It has been removed

**Comment by Camillarhi** [comment]
> @Camillarhi Can you have a look at the failing tests?

I have fixed the failing tests

**Comment by devinbinnie** [comment]
@Camillarhi I'm still seeing some failing tests, can you have a look?

**Comment by Camillarhi** [comment]
> @Camillarhi I'm still seeing some failing tests, can you have a look? Looks like some snapshots need to be updated.

Yeah, I saw that but I have updated the snapshot on my end (locally) and everything seems fine, I don't know why this keeps failing here

**Comment by Camillarhi** [comment]
> > > @Camillarhi I'm still seeing some failing tests, can you have a look? Looks like some snapshots need to be updated.
> > 
> > 
> > Yeah, I saw that but I have updated the snapshot on my end (locally) and everything seems fine, I don't know why this keeps failing here
> 
> I've seen snapshots fail due to different Node versions. Try running it under Node v18 and see if that changes anything.

I am using version 20 but I will try 18 to see if there are any changes

**Comment by Camillarhi** [comment]
@devinbinnie  I just tried 18 and I still cannot produce the error

**Comment by devinbinnie** [comment]
> @devinbinnie I just tried 18 and I still cannot produce the failed test

@Camillarhi Can you revert the changes to the snapshot? It doesn't look like it needs to be changed, it's only the ids that are mismatching

**Comment by Camillarhi** [comment]
> > @devinbinnie I just tried 18 and I still cannot produce the failed test
> 
> @Camillarhi Can you revert the changes to the snapshot? It doesn't look like it needs to be changed, it's only the ids that are mismatching

I am not sure how I can acheive that as I did not commit any changes for that file

**Comment by Camillarhi** [comment]
> > @devinbinnie I just tried 18 and I still cannot produce the failed test
> 
> @Camillarhi Can you revert the changes to the snapshot? It doesn't look like it needs to be changed, it's only the ids that are mismatching

I have reverted

File: webapp/channels/src/components/setting_item.tsx [hunk 2]
```diff
@@ -41,37 +41,42 @@ type Props = {
...
-        }
-    }
 
-    render() {
-        if (this.props.active) {
-            return this.props.max;
+        if (!active && areAllSectionsInactive) {
+            minRef.current?.focus();
         }
+    }, [active, areAllSectionsInactive]);
```
Comment (larkox) [inline webapp/channels/src/components/setting_item.tsx:63]: In order to keep the exact behavior it had before, we should remove the `areAllSectionsInactive` from the dependency list.

See that on the componentDidUpdate, we only execute when active changes (`!this.props.active && prevProps.active`). Changes in `areAllSectionsInactive` should not trigger the focus (at least according to the original implementation).

```suggestion
    }, [active]);
```

**Review by larkox — CHANGES_REQUESTED** [review]
One minor change to keep the same behavior, but apart of that looks great!

File: webapp/channels/src/components/setting_item.tsx (hunk 2 above)
Comment (Camillarhi) [reply webapp/channels/src/components/setting_item.tsx:63]: yes, I will update the dependency list of the useEffect so it is only triggered by the  `active` prop change

**Review by larkox — APPROVED** [review]

**Pull request merged**